### PR TITLE
I tried to make speed multipliers more organized.

### PR DESCRIPTION
**Pull request by TcePrepK**
The speed multipliers made by Control and code is by me.

### CODE REVIEW

**Comment by EmeraldBlock** [comment]
You may want do define the number of improvements corresponding to the number of endgame upgrades in terms of the number of endgame upgrades, instead of just 8 and 4, though that's just a suggestion.

File: src/js/game/upgrades.js [hunk 1]
```diff
@@ -22,8 +29,13 @@ function generateEndgameUpgrades() {
     }));
 }
 
-for (let i = 0; i < numEndgameUpgrades; ++i) {
-    fixedImprovements.push(0.1);
+if (numEndgameUpgrades != 0) {
+    for (let i = 0; i < 8; i++) {
+        fixedImprovements.push(0.5);
+    }
+    for (let i = 0; i < 4; i++) {
```
Comment (EmeraldBlock) [inline src/js/game/upgrades.js:36]:
```suggestion
    for (let i = 0; i < numEndgameUpgrades - 4; i++) {

```
This should make this work even if the number of upgrades changes

File: src/js/game/upgrades.js [hunk 2]
```diff
@@ -22,8 +29,13 @@ function generateEndgameUpgrades() {
     }));
 }
 
-for (let i = 0; i < numEndgameUpgrades; ++i) {
-    fixedImprovements.push(0.1);
+if (numEndgameUpgrades != 0) {
+    for (let i = 0; i < numEndgameUpgrades - 4; i++) {
```
Comment (EmeraldBlock) [inline src/js/game/upgrades.js:33]:
```suggestion
    for (let i = 0; i < 8; i++) {
```
thing

**Comment by tobspr** [comment]
Hmm, this changes the entire balancing though?

**Pull request closed without merging**